modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_compare/MiniCompare.js

```diff
@@ -92,8 +92,12 @@ function MiniCompare(props) {
 		checkConsent(COOKIE_TYPES.FUNCTIONAL)
 	);
 
+	const triggerCheckCookieConsent = () => {
+	  return !functionalCookiesConsent && items?.length > 0;
+  }
+
 	useEffect(() => {
-		if (!functionalCookiesConsent) {
+		if (triggerCheckCookieConsent()) {
```

Comment:
That condition looks also part of the "contract" between the app and the consent modal, because it short-circuits regular component rendering as well as the behavior of the effect.

So, basically, we'd need to abstract something out around these lines:

- A list of cookie types that need to be approved
- Some customizations for the cookie modal 
  - (title, alert box)
  - Success and failure toast alert texts/titles after modal is closed
- Extra, app-level conditions that must be met to check for consent instead of rendering the component (in this PR, `items > 0`). In addition, contract would ensure cookie types are approved as well.